backends/mongodb/mongodb.c

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#define BACKENDS_INTERNALS
+#include "mongodb.h"
+#include <mongoc.h>
+
+#define CONFIG_FILE_LINE_MAX ((CONFIG_MAX_NAME + CONFIG_MAX_VALUE + 1024) * 2)
+
+mongoc_client_t *mongodb_client;
+mongoc_collection_t *mongodb_collection;
+
+int mongodb_init(const char *uri_string, const char *database_string, const char *collection_string) {
+    mongoc_uri_t *uri;
+    bson_error_t error;
+
+    mongoc_init ();
+
+    uri = mongoc_uri_new_with_error (uri_string, &error);
+    if (!uri) {
+       error("BACKEND: failed to parse URI: %s. Error message: %s", uri_string, error.message);
+       return 1;
+    }
+
+    mongoc_uri_set_option_as_int32(uri, MONGOC_URI_SOCKETTIMEOUTMS, 9000); // TODO: use variable for timeout
+
+    mongodb_client = mongoc_client_new_from_uri (uri);
+    if (!mongodb_client) {
+       return 1;
+    }
+
+    mongoc_client_set_appname (mongodb_client, "netdata");
+
+    mongodb_collection = mongoc_client_get_collection (mongodb_client, database_string, collection_string);
+
+    mongoc_uri_destroy (uri);
+
+    return 0;
+}
+
+void free_bson (bson_t **insert, size_t n_documents) {
+    size_t i;
+
+    for(i = 0; i < n_documents; i++)
+        bson_destroy (insert[i]);
+
+    free(insert);
+}
+
+int mongodb_insert(char *data, size_t n_metrics) {
+    bson_t **insert = calloc(n_metrics, sizeof(bson_t *));
+    bson_error_t error;
+    char *start = data, *end = data;
+    size_t n_documents = 0;
+
+    while(*end && n_documents <= n_metrics) {
+        while(*end && *end != '\n') end++;
+
+        if(*end) {
+            *end = '\0';
+            end++;
+        }
+        else {
+            break;
+        }
+
+        insert[n_documents] = bson_new_from_json((const uint8_t *)start, -1, &error);
+
+        if (!insert[n_documents]) {
+           error("BACKEND: %s", error.message);
+           free_bson(insert, n_documents);
+           return 1;
+        }
+
+        start = end;
+
+        n_documents++;
+    }
+
+    if (!mongoc_collection_insert_many(mongodb_collection, (const bson_t **)insert, n_documents, NULL, NULL, &error)) {
+       error("BACKEND: %s", error.message);
+       free_bson(insert, n_documents);
+       return 1;
+    }
+
+    free_bson(insert, n_documents);
+
+    return 0;
+}
+
+void mongodb_cleanup() {
+    mongoc_collection_destroy (mongodb_collection);
```

Comment:
For the majority of the functions in this PR, you are using the format:

function_name(argument1, argument2);

but, for mongoc functions it looks like you are using a new pattern:

mongoc_function (argument);

It is not for all, as you can see in the 81 free_bson does not have a space between function name and argument, but in this function and others bson you are adding a space, that is not wrong, but it is not the pattern.